api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java

```diff
@@ -249,11 +259,31 @@ public Long getPhysicalNetworkId() {
             throw new InvalidParameterValueException("Unable to find network offering by ID " + networkOfferingId);
         }
 
+        Network associatedNetwork = null;
+        if (associatedNetworkId != null) {
+            associatedNetwork = _entityMgr.findById(Network.class, associatedNetworkId);
+            if (associatedNetwork == null) {
+                throw new InvalidParameterValueException("Unable to find network by ID " + associatedNetworkId);
```

Comment:
I think, this is already checked in the API call when UUID is transformed to the DB 'ID'.

Reply:
@sureshanaparti 
as discussed offline, the uuid of removed networks can be converted to long value, but the removed networks cannot be used as associated network.